babel/numbers.py

```diff
@@ -431,7 +431,7 @@ def format_compact_decimal(number, *, format_type="short", locale=LC_NUMERIC, fr
     u'123万'
     >>> format_compact_decimal(2345678, format_type="long", locale="mk")
     u'2 милиони'
-    >>> format_compact_decimal(21098765, format_type="long", locale="mk")
+    >>> format_compact_decimal(21000000, format_type="long", locale="mk")
     u'21 милион'
```

Comment:
What happened to the previous number to test with?

Reply:
It's checking if plural before rounding now so 21.0 would be singular but 21.098765 would not even if it's going to be truncated.

Assumption based on @jun66j5's comment

https://github.com/python-babel/babel/issues/931#issuecomment-1318845308